twindle-cli/index.js

```diff
@@ -9,35 +9,47 @@ const { sendToKindle } = require("./utils/send-to-kindle");
 const { getTweet } = require("./twitter-puppeteer");
 
 async function main() {
-	prepareCli();
-
-	const { format, outputFilename, tweetId, _kindleEmail, mock, shouldUsePuppeteer } = getCommandlineArgs(
-		process.argv
-	);
-
-	try {
-		// this next line is wrong
-		let tweets = require("./twitter/twitter_responses/response-version2-tweetthread.json");
-		if (!mock) {
-			if (shouldUsePuppeteer) tweets = await getTweet(tweetId);
-			else tweets = await getTweetsFromTweetId(tweetId);
-		}
-
-		const outputFilePath = getOutputFilePath(outputFilename);
-
-		await Renderer.render(tweets, format, outputFilePath);
-
-		let kindleEmail = process.env.KINDLE_EMAIL || _kindleEmail;
-		if (kindleEmail) {
-			console.devLog("sending to kindle", kindleEmail);
-			await sendToKindle(kindleEmail, outputFilePath);
-		}
-	} catch (e) {
-		console.error(e);
-	}
-
-	// If not for this line, the script never finishes
-	process.exit();
+  prepareCli();
+
+  const {
+    format,
+    outputFilename,
+    tweetId,
+    _kindleEmail,
+    mock,
+    shouldUsePuppeteer,
+  } = getCommandlineArgs(process.argv);
+
+  try {
+    // this next line is wrong
+    let tweets = require("./twitter/twitter_responses/response-version2-tweetthread.json");
+
+    if (!mock) {
+      if (shouldUsePuppeteer) tweets = await getTweet(tweetId);
+      else tweets = await getTweetsFromTweetId(tweetId);
+    }
+
+    const intelligentOutputFileName = `${
+      tweets.common.user.username
```

Comment:
@PuruVJ curious if this will work if using puppeteer

Reply:
Yeah, that part is breaking for @vijaya-bhaskar's code. 

I think we'd **have to** sync the code that puppeteer is returning. That would require hooking up with the Twitter API, or he would have to scrape that stuff himself too.

But I think we all reached a consensus that the Puppeteer code would return on IDs right? If that's finalized, we won't need to sync up any data between the 2 projects.

Reply:
For now, I will circumvent it.